components/raftstore/src/store/snapshot_backup.rs

```diff
@@ -0,0 +1,388 @@
+        Self {
+            syncer,
+            expected_epoch: None,
+            abort_when_term_change: false,
+        }
+    }
+
+    /// Create a "strict" request for waiting apply.
+    /// This will wait to last applied index, and aborts if the region epoch not
+    /// match or the last index may not be committed.
+    pub fn strict(syncer: SnapshotBrWaitApplySyncer, epoch: RegionEpoch) -> Self {
+        Self {
+            syncer,
+            expected_epoch: Some(epoch),
+            abort_when_term_change: true,
+        }
+    }
+}
+
+pub trait SnapshotBrHandle: Sync + Send + Clone {
+    fn send_wait_apply(&self, region: u64, req: SnapshotBrWaitApplyRequest) -> crate::Result<()>;
+    fn broadcast_wait_apply(&self, req: SnapshotBrWaitApplyRequest) -> crate::Result<()>;
+    fn broadcast_check_pending_admin(
+        &self,
+        tx: UnboundedSender<CheckAdminResponse>,
+    ) -> crate::Result<()>;
+}
+
+impl<EK: KvEngine, ER: RaftEngine> SnapshotBrHandle for Arc<Mutex<RaftRouter<EK, ER>>> {
+    fn send_wait_apply(&self, region: u64, req: SnapshotBrWaitApplyRequest) -> crate::Result<()> {
+        let msg = SignificantMsg::SnapshotBrWaitApply(req);
+        metrics::SNAP_BR_WAIT_APPLY_EVENT.sent.inc();
+        self.lock().unwrap().significant_send(region, msg)
+    }
+
+    fn broadcast_wait_apply(&self, req: SnapshotBrWaitApplyRequest) -> crate::Result<()> {
+        let msg_gen = || {
+            metrics::SNAP_BR_WAIT_APPLY_EVENT.sent.inc();
+            PeerMsg::SignificantMsg(SignificantMsg::SnapshotBrWaitApply(req.clone()))
+        };
+        self.lock().unwrap().broadcast_normal(msg_gen);
+        Ok(())
+    }
+
+    fn broadcast_check_pending_admin(
+        &self,
+        tx: UnboundedSender<CheckAdminResponse>,
+    ) -> crate::Result<()> {
+        self.lock().unwrap().broadcast_normal(|| {
+            PeerMsg::SignificantMsg(SignificantMsg::CheckPendingAdmin(tx.clone()))
+        });
+        Ok(())
+    }
+}
+
+#[derive(Default)]
+pub struct PrepareDiskSnapObserver {
+    before: AtomicU64,
+    initialized: AtomicBool,
+}
+
+impl PrepareDiskSnapObserver {
+    pub fn register_to(self: &Arc<Self>, coprocessor_host: &mut CoprocessorHost<impl KvEngine>) {
+        let reg = &mut coprocessor_host.registry;
+        reg.register_query_observer(0, BoxQueryObserver::new(Arc::clone(self)));
+        reg.register_admin_observer(0, BoxAdminObserver::new(Arc::clone(self)));
+        info!("registered reject ingest and admin coprocessor to TiKV.");
+    }
+
+    pub fn remained_secs(&self) -> u64 {
+        self.before
+            .load(Ordering::Acquire)
+            .saturating_sub(epoch_second_coarse())
+    }
+
+    fn reject(&self) -> CopError {
+        CopError::RequireDelay {
+            after: Duration::from_secs(self.remained_secs()),
+            reason:
+                "[Suspended] Preparing disk snapshot backup, ingests and some of admin commands are suspended."
```

Comment:
What does the `[Suspended]` refer to? If no, how about `[snapshot_backup]`? as it belongs to the feature `snapshot_backup`.

Reply:
It is just a keyword for retrying, perhaps we can remove it after check with some clients' implementation (specially `tidb-lightning`.)